app/upgrades/v5/upgrades.go

```diff
@@ -99,3 +106,50 @@ func MigrateGenesis(appState genutiltypes.AppMap, clientCtx client.Context) genu
 
 	return appState
 }
+
+// ResolveAirdrop will iterate over all the available claim records and
+// attempt to swap claimed actions and unclaimed actions.
+// The following priority is considered, and only one swap will be performed:
+// 1 - Unclaimed EVM      <-> claimed IBC
+// 2 - Unclaimed EVM      <-> claimed Vote
+// 3 - Unclaimed EVM      <-> claimed Delegate
+// 4 - Unclaimed Vote     <-> claimed IBC
+// 5 - Unclaimed Delegate <-> claimed IBC
+func ResolveAirdrop(ctx sdk.Context, k *claimskeeper.Keeper) {
+	claimsRecords := []claimstypes.ClaimsRecordAddress{}
+	k.IterateClaimsRecords(ctx, func(addr sdk.AccAddress, cr claimstypes.ClaimsRecord) (stop bool) {
+		//Perform any possible swap
+		override := swapActionsIfUncompleted(cr, claimstypes.ActionEVM, claimstypes.ActionIBCTransfer) ||
+			swapActionsIfUncompleted(cr, claimstypes.ActionEVM, claimstypes.ActionVote) ||
+			swapActionsIfUncompleted(cr, claimstypes.ActionEVM, claimstypes.ActionDelegate) ||
+			swapActionsIfUncompleted(cr, claimstypes.ActionVote, claimstypes.ActionIBCTransfer) ||
+			swapActionsIfUncompleted(cr, claimstypes.ActionDelegate, claimstypes.ActionIBCTransfer)
+
+		// If any actions were swapped, override the previous claims record
+		if override {
+			cra := claimstypes.ClaimsRecordAddress{
+				Address:                addr.String(),
+				InitialClaimableAmount: cr.InitialClaimableAmount,
+				ActionsCompleted:       cr.ActionsCompleted,
+			}
+			claimsRecords = append(claimsRecords, cra)
+		}
+		return false
+	})
+
+	for _, claim := range claimsRecords {
+		ad, _ := sdk.AccAddressFromBech32(claim.Address)
+		k.SetClaimsRecord(ctx, ad,
+			claimstypes.ClaimsRecord{InitialClaimableAmount: claim.InitialClaimableAmount,
+				ActionsCompleted: claim.ActionsCompleted})
+	}
+}
+
+func swapActionsIfUncompleted(cr claimstypes.ClaimsRecord, uncompleted, completed claimstypes.Action) bool {
```

Comment:
```suggestion
func swapUnclaimedAction(cr claimstypes.ClaimsRecord, unclaimed, claimed claimstypes.Action) bool {
```

I noticed that sometimes we use `uncompleted` and `unclaimed`. How about we use `claimed/unclaimed` everywhere, so that we only use one wording?

Reply:
+1